packaging/greenboot/microshift-running-check.sh

```diff
@@ -119,9 +119,22 @@ for i in ${!PODS_NS_LIST[@]}; do
 done
 
 # Verify that MicroShift core pods are not restarting
+declare -A pidNamespace
 for i in ${!PODS_NS_LIST[@]}; do
     CHECK_PODS_NS=${PODS_NS_LIST[$i]}
 
     echo "Checking pod restart count in the '${CHECK_PODS_NS}' namespace"
-    namespace_pods_not_restarting ${CHECK_PODS_NS}
+    namespace_pods_not_restarting ${CHECK_PODS_NS} &
+    pid=$!
```

Comment:
Is there no race-condition here for the PID assignment? The function called in the background spawns many sub-shells, so how do we know that we're getting the "right" PID?

Reply:
AFAIK `$!` is for most recent background process and we're not spawning any of those in `namespace_pods_not_restarting`. I did a small test which would suggest that we would be safe anyway:
```
+ pids=()
+ pids+=($!)
+ sleep 2
```